torch/linalg/__init__.py

```diff
@@ -140,6 +140,129 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+eigh = _add_docstr(_linalg.linalg_eigh, r"""
+linalg.eigh(input, UPLO='L') -> tuple(Tensor, Tensor)
+
+This function computes the eigenvalues and eigenvectors
+of a complex Hermitian (or real symmetric) matrix :attr:`input` or batches of such matrices
+such that :math:`\text{input} = V \text{diag}(w) V^H`, where :math:`w` is the tensor of eigenvalues,
+:math:`V` is the tensor of eigenvectors and :math:`^H` is the conjugate transpose operation.
+
+Since :attr:`input` is assumed to be consisting of Hermitian matrices,
+only the lower triangular portion of these matrices is used by default (``UPLO = 'L'``)  in computations
+and the imaginary part of the diagonal will always be treated as zero.
+
+Supports real-valued and complex-valued input.
+For complex-valued batched input on CUDA the backpropagation is not supported yet.
+
+See :func:`torch.linalg.eigvalsh` for a related function that computes only eigenvalues,
+however that function is not differentiable.
+
+.. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
+
+.. note:: The eigenvectors of matrices are not unique, any eigenvector multiplied by a constant remains
+          to be a valid eigenvector, therefore results on CPU, CUDA and from other libraries can be different.
+          Usually the difference is only in the sign of the eigenvector.
+
+.. note:: The eigenvalues/eigenvectors are computed using LAPACK/MAGMA routines ``_syevd``, ``_heevd``.
+          This function always checks whether the call to LAPACK/MAGMA is successful
+          using ``info`` argument of ``_syevd``, ``_heevd``.
+          For CUDA this causes a cross-device memory synchronization.
+
+Args:
+    input (Tensor): the Hermitian :math:`n \times n` matrix or the batch
+                    of such matrices of size :math:`(*, n, n)` where `*` is one or more batch dimensions.
+    UPLO ('L', 'U', optional): controls whether to use upper-triangular or lower-triangular part
```

Comment:
"to use upper-triangular or lower-triangular" -> "to use the upper-triangular or the lower-triangular"